kinder/pkg/build/bits/fileBits.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bits
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+
+	"k8s.io/kubeadm/kinder/pkg/extract"
+)
+
+// fileBits defines a bit installer that allows to copy a files from host machine into the node image
+type fileBits struct {
+	files []string
+}
+
+var _ Installer = &fileBits{}
+
+// NewFileBits returns a new file Installer
+func NewFileBits(files []string) Installer {
+	return &fileBits{
+		files: files,
+	}
+}
+
+// Prepare implements bits.Prepare
+func (b *fileBits) Prepare(c *BuildContext) (map[string]string, error) {
+
+	// ensure the dest path exists on host at HostBitsPath
+	dst := filepath.Join(c.HostBitsPath(), "files")
+	if err := os.Mkdir(dst, 0777); err != nil {
+		return nil, errors.Wrap(err, "failed to make bits dir")
+	}
+
+	// for each of the given file sources
+	for _, file := range b.files {
+
+		fileSrcDst := strings.Split(file, ":")
+
+		if len(fileSrcDst) != 2 {
+			return nil, errors.New("invalid value for --with-file option")
```

Comment:
what about adding the wrong file name to the error?

Reply:
Destination directory must be an absolute path, right? what if user input a relative path? thoughts on maybe adding a check to parse if the destination directory starts from a "/" or not?

Reply:
done